File: client/src/app/models/apartment.ts
export interface Apartment {
  id: number;
  name: string;
  description: string;
  price_per_night: number;  
  capacity: number;
  bedrooms: number;
  bathrooms: number;
  image_url: string;  
  images?: string[];
  amenities: string[];
  address: string;
  city: string;
  rating?: number;
  created_at?: string;
  updated_at?: string;
}

File: greyhill-backend/app/models/apartment.py
from sqlalchemy import Column, Integer, String, Float, JSON, DateTime
from sqlalchemy.sql import func
from app.database import Base

class Apartment(Base):
    __tablename__ = "apartments"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(200), nullable=False)
    description = Column(String(1000))
    price_per_night = Column(Float, nullable=False)
    capacity = Column(Integer, nullable=False)
    bedrooms = Column(Integer, nullable=False)
    bathrooms = Column(Integer, nullable=False)
    image_url = Column(String(500))
    images = Column(JSON)
    amenities = Column(JSON)
    address = Column(String(300))
    city = Column(String(100))
    rating = Column(Float, default=0.0)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())